source/code-snippets/usage-examples/bulkWrite.ts

```diff
@@ -0,0 +1,53 @@
+import { MongoClient } from "mongodb";
+
+// Replace the uri string with your MongoDB deployment's connection string.
+const uri =
+  "mongodb+srv://<user>:<password>@<cluster-url>?writeConcern=majority";
+
+const client = new MongoClient(uri);
+
+async function run() {
+  try {
+    await client.connect();
+
+    const database = client.db("sample_mflix");
+    const theaters = database.collection("theaters");
```

Comment:
I think it may also be worthwhile to add a type to this collection, because the insert/update operations is where typescript is actually helpful in terms of hinting which properties are expected on an object

Reply:
Added  a type to the collection and I think it improves the UX! Good catch. 

I also change fields specified like:

> `field.subfield.another_subfield : "value"`

To:

> `{field : {subfield : { another_subfield : "value"}}}`

As typescript did not seem to able to determine the correct type of the field with the first syntax.

Reply:
Unfortunately there are no typescript solutions for this that would not result in unacceptably long compilation times for our users' projects. We should continue advocating for using dot notation in queries, just warn about the inevitable loss of type safety.

Reply:
Sounds good, thanks for looking into this. I'll add a note under the example warning about loss of type safety with dot notation.